src/website/shared/auth.py

```diff
@@ -0,0 +1,143 @@
+import logging
+from typing import Any, cast
+
+from allauth.socialaccount.models import SocialAccount
+from django.contrib.auth.models import Group, Permission, User
+from django.db.models import Q
+from github import Github
+from github.NamedUser import NamedUser
+from github.Organization import Organization
+from github.Team import Team
+
+from shared.utils import get_gh
+
+github: Github = get_gh(per_page=100)  # 100 is the API limit
+logger = logging.getLogger(__name__)
+
+
+def get_gh_username(user: User) -> str | None:
+    """
+    Return the Github username of a given Auth.User.
+    """
+    social_user = User.objects.get(id=user.id)  # type: ignore
+    social_account: SocialAccount | None = (
+        social_user.socialaccount_set.filter(provider="github").first()  # type: ignore
+    )
+    if social_account:
+        return social_account.extra_data.get("login")  # type: ignore
```

Comment:
Sounds like what you care about is `.values_list('extra_data__login')` or something like that, no? You could get rid of the many attributes that could get passed via the SQL query.

Reply:
Doing
```
def get_gh_username(user: User) -> str | None:
    """
    Return the Github username of a given Auth.User.
    """
    social_user = User.objects.get(id=user.id)  # type: ignore
    try: 
        extra_login: str = (
            social_user.socialaccount_set.filter(provider="github")
                .values_list('extra_data__login', flat=True)[0]
        )
        return extra_login
    except Exception:
        logger.exception("Failed to get GitHub username for user %s.", user)
        return None
```
Gives me 
```
django.core.exceptions.FieldError: Unsupported lookup 'login' for JSONField or join on the field not permitted.
```

For now I'll keep the `extra_data.get("login")` and give this another spin later.